packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -0,0 +1,269 @@
+  /// [IconTheme], if any.
+  ///
+  /// If no [IconTheme]s are specified, icons will default to black.
+  ///
+  /// In material apps, if there is a [Theme] without any [IconTheme]s
+  /// specified, icon colors default to white if the theme is dark
+  /// and black if the theme is light.
+  /// See [Theme] to set the current theme and [ThemeData.brightness]
+  /// for setting the current theme's brightness.
+  final Color color;
+
+  /// The size of the icon in logical pixels.
+  ///
+  /// Icons occupy a square with width and height equal to size.
+  ///
+  /// Defaults to the current [IconTheme] size.
+  final double size;
+
+  /// The icon to display. Available icons are listed in [AnimatedIcons].
+  final AnimatedIconData icon;
+
+  /// Semantic label for the icon.
+  ///
+  /// This would be read out in accessibility modes (e.g TalkBack/VoiceOver).
+  /// This label does not show in the UI.
+  ///
+  /// See also:
+  ///
+  ///  * [Semantics.label], which is set to [semanticLabel] in the underlying
+  ///    [Semantics] widget.
+  final String semanticLabel;
+
+  /// The text direction to use for rendering the icon.
+  ///
+  /// If this is null, the ambient [Directionality] is used instead.
+  ///
+  /// If the text diection is [TextDirection.rtl], the icon will be mirrored
+  /// horizontally (e.g back arrow will point right).
+  final TextDirection textDirection;
+
+  static final _UiPathFactory _pathFactory = () => new ui.Path();
+
+  @override
+  Widget build(BuildContext context) {
+    final _AnimatedIconData iconData = icon;
+    final IconThemeData iconTheme = IconTheme.of(context);
+    final double iconSize = size ?? iconTheme.size;
+    final TextDirection textDirection = this.textDirection ?? Directionality.of(context);
+    return new Semantics(
+      label: semanticLabel,
+      child: new CustomPaint(
+        size: new Size(iconSize, iconSize),
+        painter: new _AnimatedIconPainter(
+          iconData.paths,
+          progress,
+          color ?? iconTheme.color,
+          iconSize / iconData.size.bottomRight(const Offset(0.0, 0.0)).dx,
+          textDirection == TextDirection.rtl,
+          _pathFactory,
+        ),
+      ),
+    );
+  }
+}
+
+typedef ui.Path _UiPathFactory();
+
+class _AnimatedIconPainter extends CustomPainter {
+  _AnimatedIconPainter(
+    this.paths,
+    this.progress,
+    this.color,
+    this.scale,
+    this.shouldMirror,
+    this.uiPathFactory,
+  ) : super(repaint: progress);
+
+  // This list is assumed to be immutable, changes to the contents of the list
+  // will not trigger a redraw as shouldRepaint will keep returning false.
+  final List<_PathFrames> paths;
+  final Animation<double> progress;
+  final Color color;
+  final double scale;
+  /// If this is true the image will be mirrored horizontally.
+  final bool shouldMirror;
+  final _UiPathFactory uiPathFactory;
+
+  @override
+  void paint(ui.Canvas canvas, Size size) {
+    canvas.scale(scale, scale);
```

Comment:
add a comment saying that the CustomPaint widget implies a save/restore around this call, otherwise every time I see this method I'll twitch thinking we're leaking the transform. :-)

Reply:
done